### ISSUE TITLE
Invalid level of header

### ISSUE THREAD

**Issue by kongou-ae** [redpen-cc/redpen#636]
I tested this markdown using this javascript validator for getting the content of header.

```markdown
# 1

ああああああああああああああ

## 1.1

あああああああああああああ

### 1.1.1

あああああああああああああ

# 2

ああああああああああああああ

## 2.1

ああああああああああ。
```

```javascirpt
function validateSection(section){
    console = {
        log:print
    };


    console.log(section.getLevel())
    console.log(section.getHeaderContent(0))
}
```

The result of this test was as below. What is the section of level 0 ?

```
0
Sentence{links=[], lineNumber=0, content='', startPositionOffset=0, isFirstSentence=false, tokens=[], offsetMap=[]}
1
Sentence{links=[], lineNumber=1, content='1', startPositionOffset=2, isFirstSentence=true, tokens=[], offsetMap=[LineOffset(1, 2)]}
2
Sentence{links=[], lineNumber=5, content='1.1', startPositionOffset=3, isFirstSentence=true, tokens=[], offsetMap=[LineOffset(5, 3), LineOffset(5, 4), LineOffset(5, 5)]}
3
Sentence{links=[], lineNumber=9, content='1.1.1', startPositionOffset=4, isFirstSentence=true, tokens=[], offsetMap=[LineOffset(9, 4), LineOffset(9, 5), LineOffset(9, 6), LineOffset(9, 7), LineOffset(9, 8)]}
1
Sentence{links=[], lineNumber=13, content='2', startPositionOffset=2, isFirstSentence=true, tokens=[], offsetMap=[LineOffset(13, 2)]}
2
Sentence{links=[], lineNumber=17, content='2.1', startPositionOffset=3, isFirstSentence=true, tokens=[], offsetMap=[LineOffset(17, 3), LineOffset(17, 4), LineOffset(17, 5)]}
```

**Comment by takahi-i** [comment]
Thank you very much for telling us the wired behavior in RedPen, @kongou-ae! We will fix the behavior.

**Comment by takahi-i** [comment]
I thought this is a bug but now I remember that adding **level 0 section** by the Markdown parser is an expected behavior.

The level 0 section in the beginning of the document is added to handle the sentences before the first section.

For example, there is a document as follows.

     this sentence exist before the first section.

     # section 1
     blah blah

     # section 2

**Comment by takahi-i** [comment]
I agree that the behavior is a bit wired as you point out, and therefore we should remove the level zero section 0 when there is no sentences before the first section. Any comments would be appreciate it.

**Comment by kongou-ae** [comment]
@takahi-i , Thank you for your detailed explanation. I understood the reason of the level zero section 0. I am looking forward to be implemented #648.

**Comment by takahi-i** [comment]
I merged the fix for this problem in #648. Hopefully the fix works for you. Let me know when the fix does not work adding another issue or reopen this issue.
